### PR TITLE
add support for debian bullseye

### PR DESCRIPTION
**Pull request by duviful**
Add support for Debian bullseye, which doesn't have python3.8 (only 3.9).
Bumped ansible-role-postgresql version in molecule if we want at some point add debian in tests. This version comes with support for Debian-11.

Feel free to modify the code to adapt to your coding style.

### CODE REVIEW

File: tasks/variables.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+# Variable configuration.
+- name: Include OS-specific variables (Debian).
+  ansible.builtin.include_vars: "{{ ansible_distribution }}-{{ ansible_distribution_version.split('.')[0] }}.yml"
+  when: ansible_os_family == 'Debian'
+
+- name: Define python_packages.
```
Comment (jvoss) [inline tasks/variables.yml:7]: This task winds up being triggered when not Debian (Ubuntu and RHEL) as well and failing.

I like the method you are using here however either all supported distros will need refactored to this style or you will need to add a conditional here for Debian as well.

I am also happy if you add a conditional here for now and want me to refactor the rest.

File: tasks/variables.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  ansible.builtin.set_fact:
+    python_packages: "{{ __python_packages | list }}"
+  when: python_packages is not defined
```
Comment (jvoss) [inline tasks/variables.yml:10]: Temporary workaround (see note above):
```
when:
  - ansible_os_family == 'Debian'
  - python_packages is not defined

**Review by jvoss — CHANGES_REQUESTED** [review]
Thank you for adding Debian support! I like the progress however there is one item to address that is breaking installation on other distributions.

**Comment by duviful** [comment]
I'm also adding some more variables to have `tasks/upgrade.yml` properly working with the changes I've introduced.

**Comment by duviful** [comment]
my local molecule fails at

```
TASK [geerlingguy.postgresql : Ensure PostgreSQL is started and enabled on boot.] ***
fatal: [instance]: FAILED! => {"changed": false, "cmd": "/bin/systemctl", "msg": "Failed to connect to bus: No such file or directory", "rc": 1, "stderr": "Failed to connect to bus: No such file or directory\n", "stderr_lines": ["Failed to connect to bus: No such file or directory"], "stdout": "", "stdout_lines": []}
```

do you use it for test or you moved to github actions fro testing?

**Comment by jvoss** [comment]
> my local molecule fails at
> 
> ```
> TASK [geerlingguy.postgresql : Ensure PostgreSQL is started and enabled on boot.] ***
> fatal: [instance]: FAILED! => {"changed": false, "cmd": "/bin/systemctl", "msg": "Failed to connect to bus: No such file or directory", "rc": 1, "stderr": "Failed to connect to bus: No such file or directory\n", "stderr_lines": ["Failed to connect to bus: No such file or directory"], "stdout": "", "stdout_lines": []}
> ```
> 
> do you use it for test or you moved to github actions fro testing?

I also have had a lot of problems with `geerlingguy.postgresql` for local testing as well, at least on OSX. I'm curious if developing on Linux has the same issue considering the GitHub actions run fine.

I have been relying more on testing during the PR checks or manually with a test VM but I too wish I could rely on local molecule testing.

File: tasks/variables.yml (hunk 2 above)
Comment (jvoss) [reply tasks/variables.yml:20]: Thanks for adding the other distro variables!

File: tasks/variables.yml (hunk 1 above)
Comment (jvoss) [reply tasks/variables.yml:17]: Thank you very much for this!

**Review by jvoss — CHANGES_REQUESTED** [review]
I think we are getting closer. There is a potential order of operations issue that is causing some of the builds to fail (see review comments).

Feel free to add the appropriate Debian distros the GitHub actions matrices to include in testing if you like. Otherwise I will be adding them at the completion of this PR to ensure we do not break it in the future.

File: vars/Amazon-2.yml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+---
+__python_version: "3.8"
+__python_packages: "python{{ python_version }}"
```
[the remark this answers is not in the document]
Comment (duviful) [reply vars/Amazon-2.yml:3]: I think you are right and I like the solution you provided.

**Comment by jvoss** [comment]
> I'm sorry for this trial and error approach but without proper testing, i' just trying...

It's no problem at all! This is what PRs are for. Feel free to send as many commits as you need.

**Comment by duviful** [comment]
I moved my local molecule testing on a amd64 machine and I'm using molecule-vagrant. 
I added boxes for all the OS flavors.
I still got some non-zero return codes during configure.py to sort out.

For example, debian 10 (on python 3.7) gets a warning from netbox 3.11 that will not be supported in the future. this is interpreted as a non zero by ansible and fails the task.

**Review by jvoss — APPROVED** [review]

**Pull request merged**